tests/core/test_lightning_optimizer.py

```diff
@@ -433,3 +435,175 @@ def configure_optimizers(self):
 
         assert adam_zero_grad.call_count == 20
         assert sgd_zero_grad.call_count == 5
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="test requires GPU machine")
+@pytest.mark.parametrize(["precision", "amp_backend"], [
+    pytest.param(16, "native"),
+    pytest.param(32, "native"),
+])
+@pytest.mark.parametrize('accumulate_grad_batches', [1])
+def test_parity_training_lightning_optimizer(tmpdir, amp_backend, precision, accumulate_grad_batches):
+    """
+    Test training with accumulated gradients with and within enable_pl_optimizer reaches the same weights
+    """
+
+    initial_weights = {}
+
+    class TestBoringModel(BoringModel):
+
+        def __init__(self, optimizer_name):
+            super().__init__()
+            self._optimizer_name = optimizer_name
+            self.losses = []
+            self.grads = []
+            self.on_before_zero_grad_count = 0
```

Comment:
would it be better to mock it?
https://medium.com/python-pandemonium/python-mocking-you-are-a-tricksy-beast-6c4a1f8d19b2
but maybe not needed now...

Reply:
Could do, but I am also capture the gradients to make sure everything is properly designed.